internals/overlord/overlord.go

```diff
@@ -318,15 +320,15 @@ func (o *Overlord) settle(timeout time.Duration, beforeCleanups func()) error {
 		if timeout > 0 && time.Since(t0) > timeout {
 			err := fmt.Errorf("Settle is not converging")
 			if len(errs) != 0 {
-				return &ensureError{append(errs, err)}
+				return &multiError{append(errs, err)}
 			}
 			return err
 		}
 		next := o.ensureTimerReset()
 		err := o.stateEng.Ensure()
 		switch ee := err.(type) {
 		case nil:
-		case *ensureError:
+		case *multiError:
```

Comment:
Major difference here: "is it an error in ensure" vs. "is this any sort of error that happens to group multiple errors". Doesn't matter for the use case here, but is there any other where it actually does?

Reply:
As I indicated in the comment above, I haven't seen any logic in Pebble that handles the various error types in a different way.